spock-core/src/main/java/org/spockframework/runtime/model/FeatureInfo.java

```diff
@@ -34,6 +35,10 @@ public class FeatureInfo extends SpecElementInfo<SpecInfo, AnnotatedElement> imp
   private final List<DataProviderInfo> dataProviders = new ArrayList<>();
   private final IterationFilter iterationFilter = new IterationFilter();
 
+  private final List<FeatureInfo> dependencies = new ArrayList<>();
+
+  private FeatureNode node;
```

Comment:
This introduces a cyclic dependency between these classes.

Reply:
And would that be a problem?